lua/neo-tree/sources/filesystem/lib/fs_scan.lua

```diff
@@ -9,13 +10,17 @@ local log = require("neo-tree.log")
 local fs_watch = require("neo-tree.sources.filesystem.lib.fs_watch")
 local git = require("neo-tree.git")
 
+local Path = require "plenary.path"
+local os_sep = Path.path.sep
+
 local M = {}
 
-local dir_complete = function(context, dir_path)
+local on_directory_loaded = function(context, current_dir)
   local state = context.state
-  local paths_to_load = context.paths_to_load
-  local folders = context.folders
-
+  local scanned_folder = context.folders[current_dir]
+  if scanned_folder then
+    scanned_folder.loaded = true
+  end
   if state.use_libuv_file_watcher then
     local root = context.folders[dir_path]
```

Comment:
I think `dir_path` needs to be updated to `current_dir`. You'll need to test with the `use_libuv_file_watcher = true` to make sure it is working.

Reply:
good catch, thanks